ReactSkia/views/common/RSkImageCacheManager.cpp

```diff
@@ -0,0 +1,144 @@
+  #ifdef RNS_SHELL_HAS_GPU_SUPPORT
+    GrDirectContext* gpuContext =RSkSurfaceWindow::getDirectContext();
+    if(gpuContext) {
+      gpuContext->getResourceCacheUsage(&fOldCount, &gpuCacheUsed);       
+    }
+  #endif
+    cpuCacheUsed = SkGraphics::GetResourceCacheTotalBytesUsed();
+    RNS_LOG_DEBUG("CPU CACHE consumed bytes: "<<cpuCacheUsed<< ",, GPU CACHE consumed bytes: "<<gpuCacheUsed);
+
+    //Evict entry on reaching HWM. This logic to be enchanced based on growing need and clarity.
+    for(it = ImageCacheMap.begin();(it != ImageCacheMap.end()) && (evictCount < EVICT_COUNT);) {
+      if((cpuCacheUsed < SKIA_CPU_IMAGE_CACHE_HWM_LIMIT) && ( gpuCacheUsed < SKIA_GPU_IMAGE_CACHE_HWM_LIMIT))
+        break;
+      if((it->second)->unique()) {
+          ImageCacheMap.erase(it++);
+          evictCount++;
+          RNS_LOG_DEBUG("EVICTING Entry for RNS Image Hash Map ...");
+      } else {
+      	  ++it;
+      }
+    }
+    return ((cpuCacheUsed < SKIA_CPU_IMAGE_CACHE_LIMIT) && ( gpuCacheUsed < SKIA_GPU_IMAGE_CACHE_LIMIT));
+  }
+
+}//namespace
+
+sk_sp<SkImage> getImageData(const char *path) {
+
+  hash<string> hashedKey; 
+  sk_sp<SkImage> imageData{nullptr};
+  //Set Cache limit, if not set before
+  if(cpuCacheLimit_ != SKIA_CPU_IMAGE_CACHE_LIMIT) {
+    setCpuImageCacheLimit(SKIA_CPU_IMAGE_CACHE_LIMIT);
+  }
+#ifdef RNS_SHELL_HAS_GPU_SUPPORT
+  if(gpuCacheLimit_ != SKIA_GPU_IMAGE_CACHE_LIMIT) {
+    setGpuImageCacheLimit(SKIA_GPU_IMAGE_CACHE_LIMIT);
+  }
+#endif
+  if(!path) {
+    RNS_LOG_ERROR("Invalid File");
+	  return nullptr;
+  }
+ //first check file entry in hash map, if entry not exist, create imageData
+  map<size_t, sk_sp<SkImage>>::iterator it = ImageCacheMap.find(hashedKey(path));
+  imageData= ((it != ImageCacheMap.end()) ? it->second : nullptr);
+  if(!imageData) {
+    imageData = makeImageData(path);
+    //Add entry to hash map only if the cache mem usage is with in the limit
+    if(evictAsNeeded() && imageData) {
+      ImageCacheMap.insert(pair<size_t, sk_sp<SkImage>>(hashedKey(path),imageData));
+      RNS_LOG_DEBUG("New Entry in Map..."<< ImageCacheMap.size()<<"for file :"<<path);
+    } else {
+        RNS_LOG_DEBUG("mem limit Reached , file is not cached ...");
+    }
+  }
+  return imageData;
+}
+
+#ifdef RNS_IMAGE_CACHE_USAGE_DEBUG
+void printCacheUsage() {
+  size_t cpuUsedMem{0},gpuUsedMem{0};
+  static size_t prevCpuUsedMem{0},prevGpuUsedMem{0};
+  int fOldCount{0};
+
+  cpuUsedMem = SkGraphics::GetResourceCacheTotalBytesUsed(); 
+#ifdef RNS_SHELL_HAS_GPU_SUPPORT
+  GrDirectContext* gpuContext =RSkSurfaceWindow::getDirectContext();
+  if(gpuContext) {
+    gpuContext->getResourceCacheUsage(&fOldCount, &gpuUsedMem);
+    RNS_LOG_INFO("Memory consumed for this run in GPU CACHE:"<<(gpuUsedMem - prevGpuUsedMem));
+  }
+#endif
+  RNS_LOG_INFO("Memory consumed for this run in CPU CACHE :"<<(cpuUsedMem - prevCpuUsedMem));
+  prevCpuUsedMem = cpuUsedMem;
+  prevGpuUsedMem = gpuUsedMem;
```

Comment:
Can prevGpuUsedMem assignment brought under *_GPU_SUPPORT macro ?
This is to avoid one instruction for non-GPU targets.
Also applicable for the variables initialized in beginning.